src/app/packages/core/utils/limited-array.ts

```diff
@@ -0,0 +1,45 @@
+import { cloneDeep } from 'lodash';
+
+export interface LimitedArray {
+	data: Array<any>;
+	limited: number;
+}
+
+// input: array, number and (*sorting function)
+// output: LimitedArray object with data property (array values up to limit) and number (how many data cells where
+// sliced off thr original array)
+// when (*sorting function) provided, sorting will be perform before slice
+export function limitArray(arr: Array<any>, limit: number, sortFn?: (a: any, b: any) => number): LimitedArray {
+	const limited = (limit < arr.length) ? arr.length - limit : 0;
+	let data = cloneDeep(arr);
+	if (sortFn) {
+		data = data.sort(sortFn);	// sort before limit
+	}
+	if (limited > 0) {
+		data = data.slice(0, limit);	// slice the part of the array beyond limit value
+	}
+	return { data, limited }
+}
+
+// input: array of LimitedArray objects, number and (*sorting function)
+// output: LimitedArray object with data property (merged array from all input arrays up to limit) and number (how many data cells where
+// sliced off the merged array)
+// when (*sorting function) provided, sorting will be perform before slice
+export function mergeLimitedArrays(arr: Array<LimitedArray>, limit: number, sortFn?: (a: any, b: any) => number): LimitedArray {
+	let	data = [];
+	let	limited = 0;
+	arr.forEach(limitedArr => {
+		data = data.concat(limitedArr.data);
+		limited += limitedArr.limited;
+	});
+
+	const mergedLimited = (limit < data.length) ? data.length - limit : 0;
+	if (sortFn) {
+		data = data.sort(sortFn);	// sort before limit
```

Comment:
The sort method sorts the Array object in place; no new Array object is created during execution.

Reply:
This is merged array. it has to be re-sorted (in case specific order is require, which in this case it does)

Reply:
read again

Reply:
OK, will do